homeassistant/components/vacuum/__init__.py

```diff
@@ -119,7 +121,7 @@ def is_on(hass: HomeAssistant, entity_id: str) -> bool:
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the vacuum component."""
-    component = hass.data[DOMAIN] = EntityComponent(
+    component = hass.data[DOMAIN] = EntityComponent[VacuumEntity](
```

Comment:
```suggestion
    component = hass.data[DOMAIN] = EntityComponent[_BaseVacuum](
```
This should probably be `_BaseVacuum` instead. `StateVacuumEntity` doesn't inherit from `VacuumEntity`.